connectors/src/connectors/zendesk/index.ts

```diff
@@ -215,46 +232,37 @@ export class ZendeskConnectorManager extends BaseConnectorManager<null> {
 
   /**
    * Launches a full re-sync workflow for the connector,
-   * restarting workflows with the signals necessary to resync every resource selected by the user.
-   * It sends signals for all the brands and for all the categories whose Help Center is not selected as a whole.
+   * syncing every resource selected by the user with forceResync = true.
    */
-  async sync(): Promise<Result<string, Error>> {
+  async sync({
+    fromTs,
+  }: {
+    fromTs: number | null;
+  }): Promise<Result<string, Error>> {
     const { connectorId } = this;
     const connector = await ConnectorResource.fetchById(connectorId);
     if (!connector) {
       logger.error({ connectorId }, "[Zendesk] Connector not found.");
       return new Err(new Error("Connector not found"));
     }
 
-    // syncing all the brands syncs tickets and whole help centers when selected
-    const brandIds = await ZendeskBrandResource.fetchAllBrandIds(connectorId);
-    const noReadHelpCenterBrandIds =
-      await ZendeskBrandResource.fetchHelpCenterReadForbiddenBrandIds(
-        connectorId
-      );
-    // syncing individual categories syncs for ones where the Help Center is not selected as a whole
-    const categoryIds = (
-      await concurrentExecutor(
-        noReadHelpCenterBrandIds,
-        async (brandId) => {
-          const categoryIds =
-            await ZendeskCategoryResource.fetchReadOnlyCategoryIdsByBrandId({
-              connectorId,
-              brandId,
-            });
-          return categoryIds.map((categoryId) => ({ categoryId, brandId }));
-        },
-        { concurrency: 10 }
-      )
-    ).flat();
-
-    const result = await launchZendeskSyncWorkflow(connector, {
-      brandIds,
-      categoryIds,
-      forceResync: true,
-    });
+    // launching an incremental workflow taking the diff starting from the given timestamp
+    if (fromTs) {
+      const cursors = await ZendeskTimestampCursor.findOne({
+        where: { connectorId },
+      });
+      if (!cursors) {
+        throw new Error("[Zendesk] Timestamp cursor not found.");
```

Comment:
nit: error could be clearer  ("Cannot full resync a zensdesk connector that has never completed initial sync" or something) + don't you want to delete the cursor if fromTs is null ?

Reply:
i.e same code path for both cases, but if !!fromTs -> update cursor.ts to fromTs, otherwise delete cursord?

Reply:
true, let's delete the cursor if `!fromTs`

Reply:
So with `fromTs`,  `forceResync` doesn't make a lot of sense I guess

Reply:
as per IRL: `!!fromTs` -> launch incremental sync from `fromTs` without `forceResync`, otherwise launch full resync with `forceResync` to true.
No code path to launch full resync without `forceResync` -> can be done using the CLI